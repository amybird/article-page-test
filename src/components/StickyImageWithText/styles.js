import styled from 'styled-components';
import mq from '../../styles/mediaQueries';

export const ImageWrap = styled.div`
  width: 100%;
  top: 5px;
  display: inline-block;

  ${mq.medium`
    margin-top: 70px;
    padding: 20px;
    position: sticky;
    width: 50%;
    vertical-align: top;
  `}
`;

export const Paragraphs = styled.div`
  width: 100%;
  display: inline-block;
  padding: 20px 10px;

  ${mq.medium`
    width: 50%;
  `}
`;
