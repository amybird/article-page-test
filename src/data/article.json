{
    "articleName": "Standing strong",
    "articleType": "Cover Story",
    "featuring": "Danai Gurira",
    "content": [
        {
            "contentType":"HeroImage",
            "content":{
                "images":{
                    "small":"https://cache.net-a-porter.com/content/images/story-head-content-SUBBED-1556275864012.jpeg/w600_q65.jpeg",
                    "large": "https://cache.net-a-porter.com/content/images/story-head-content-SUBBED-1556275864012.jpeg/w1900_q65.jpeg"
                },
                "text":{
                    "articleType": "Cover Story",
                    "articleName": "Standing strong",
                    "featuring": "Danai Gurira"
                }
            }
        },
        {
            "contentType":"StickyImageWithText",
            "content":{
                "imageLeft": false,
                "paragraphs": [
                    "Going unnoticed is not in Danai Gurira’s repertoire. When the 41-year-old actress walks into the roof restaurant of the Dream hotel in Hollywood, wearing faded denim, a cream silk blouse and mirrored aviators, she practically gleams in the morning light. It’s hard to say if the other patrons recognize her as General Okoye, the fierce protector of the Wakanda royal family in Black Panther, or the equally imposing zombie-slayer Michonne from The Walking Dead, or are just struck by the way she carries herself, but all eyes are on her.",
                    "“I’m running a writers’ room near here,” Gurira says, sitting down and peering over the railing at the sprawl of surrounding offices and studios. She is talking about the newest line on her résumé: head writer for a television mini-series. Aside from her commanding performances on screen, she’s also a feted writer whose plays have garnered multiple awards, including a Tony nomination. Now she’s adapting Chimamanda Ngozi Adichie’s acclaimed novel Americanah – a love story following a Nigerian couple through their complicated lives – with her longtime friend Lupita Nyong’o. The two became close in 2015 when Nyong’o starred in Gurira’s all-female play Eclipsed, about the Liberian civil war."  
                ],
                "quote": "“It’s about collaborating NOT cooperating. I’m pretty willing to have a CLEAR conversation about what I think WORKS, and what doesn’t, and why”",
                "image":"https://cache.net-a-porter.com/content/images/story-body-content-SUBBED-0-1556266075382.jpeg/w950_q65.jpeg"
            }
        },
        {
            "contentType":"StickyImageWithText",
            "content":{
                "imageLeft": true,
                "paragraphs": [
                    "Gurira is tight-lipped about her upcoming departure from The Walking Dead, for which she’s scheduled to guest star in a handful of episodes in the tenth season. “We start shooting soon, so I’ll have to start fitting in my workouts,” she says of her Facetime sessions with her trainer. In a month, she’ll return to Atlanta, now practically her third home – as the location for the long-running zombie series as well as the Marvel shoots – alongside her apartment in Brooklyn and home in Hollywood. “I have a really beautiful family in The Walking Dead,” is what she tells me. “We have pot luck dinners. We go to each other’s homes. We check in with each other. I also talk a lot to my show runner, Scott Gimple, who is kind of my ‘Bat phone’ as I navigate running a writer’s room.”",
                    "For the task of adapting Americanah, Gurira is actually uniquely qualified. “Funnily, me and Chimamanda have these interesting parallels,” she says. “We were both raised by academics, either near or on an African university campus. We saw our countries go through various things when we were growing up. And we came back to America for school. Lupita, too.”",
                    "Born to African parents in the US, Gurira returned to her family’s native Zimbabwe when she was five, and then came back to the States for college in Minnesota. “I am definitely a strong meshing of both worlds,” she says, explaining that she feels American and Zimbabwean in equal parts. She first ‘noticed’ her cultural composition when she was a student at Macalester College in Saint Paul, Minnesota. “I knew when I got here that I had to be vigilant. Being in a land of such access and excess, you have to really be clear about what you’re doing and why, and how you’re going to be responsible with all of the opportunities available to you as an American citizen. And it was clear to me that I was going to retain a tangible and active connection to Zimbabwe. When I moved to Africa as a girl, I thought I was American. Then when I came back, I realized how African I was.”"  
                ],
                "quote": "“I am a strong meshing of BOTH worlds. When I moved to Africa as a girl, I thought I was AMERICAN. Then when I came back, I realized how AFRICAN I was”",
                "image":"https://cache.net-a-porter.com/content/images/story-body-content-SUBBED-0-1556276906399.jpeg/w950_q65.jpeg"
            }
        },
        {
            "contentType":"Paragraph",
            "content":{
                "paragraphs": [
                    "Going unnoticed is not in Danai Gurira’s repertoire. When the 41-year-old actress walks into the roof restaurant of the Dream hotel in Hollywood, wearing faded denim, a cream silk blouse and mirrored aviators, she practically gleams in the morning light. It’s hard to say if the other patrons recognize her as General Okoye, the fierce protector of the Wakanda royal family in Black Panther, or the equally imposing zombie-slayer Michonne from The Walking Dead, or are just struck by the way she carries herself, but all eyes are on her.",
                    "“I’m running a writers’ room near here,” Gurira says, sitting down and peering over the railing at the sprawl of surrounding offices and studios. She is talking about the newest line on her résumé: head writer for a television mini-series. Aside from her commanding performances on screen, she’s also a feted writer whose plays have garnered multiple awards, including a Tony nomination. Now she’s adapting Chimamanda Ngozi Adichie’s acclaimed novel Americanah – a love story following a Nigerian couple through their complicated lives – with her longtime friend Lupita Nyong’o. The two became close in 2015 when Nyong’o starred in Gurira’s all-female play Eclipsed, about the Liberian civil war."  
                ]
            }
        }
    ]
}
