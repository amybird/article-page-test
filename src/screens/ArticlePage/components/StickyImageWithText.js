import React from 'react';
import Paragraph from '../../../ui/Paragraph';
import ResponsiveImage from '../../../ui/ResponsiveImage';
import Container from '../../../ui/Container';
import mq from '../../../styles/mediaQueries';
import styled, {css} from 'styled-components/macro';
import Quote from '../../../ui/Quote';
import shortId from 'shortid';

class StickyImageWithText extends React.PureComponent { 

    render() {
       const { image, paragraphs, quote, imageLeft } = this.props[0];
       
        return (
            <Container my={20}>
                    {imageLeft && <ImageWrap><ResponsiveImage smallImg={image}/></ImageWrap>}
                    <Paragraphs>
                        {paragraphs.map( (paragraph, index)=> 
                            <Paragraph key={shortId.generate() } largeFirstLetter={index === 0 ? true : false}>{paragraph}</Paragraph>
                        )}
                        <Quote quote={quote}/>
                    </Paragraphs>
                    {!imageLeft && <ImageWrap><ResponsiveImage smallImg={image}/></ImageWrap>}         
            </Container>
        )
    }
}

const ImageWrap = styled.div`
    width:100%;
    top:5px;
    display:inline-block;

    ${mq.medium(css`
        margin-top:70px;
        padding: 20px;
        position:sticky;
        width:50%;
        vertical-align: top;
    `)}
`;

const Paragraphs = styled.div`
    width:100%;
    display:inline-block;
    padding: 20px 10px;

    ${mq.medium(css`
        width:50%;
    `)}
`;

export default StickyImageWithText;