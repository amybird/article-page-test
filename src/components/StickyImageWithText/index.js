import React from 'react';
import shortId from 'shortid';
import PropTypes from 'prop-types';
import Paragraph from '../Paragraph';
import ResponsiveImage from '../ResponsiveImage';
import Container from '../Container';
import Quote from '../Quote';
import { ImageWrap, Paragraphs } from './styles';

const StickyImageWithText = ({ data }) => {
  const {
    image, paragraphs, quote, imageLeft
  } = data.content;

  return (
    <Container>
      {imageLeft && (
      <ImageWrap>
        <ResponsiveImage smallImg={image} />
      </ImageWrap>
      )}
      <Paragraphs>
        {paragraphs.map((paragraph, index) => (
          <Paragraph
            key={shortId.generate()}
            largeFirstLetter={index === 0}
          >
            {paragraph}
          </Paragraph>
        ))}
        <Quote text={quote} />
      </Paragraphs>
      {!imageLeft && (
      <ImageWrap>
        <ResponsiveImage smallImg={image} />
      </ImageWrap>
      )}
    </Container>
  );
};

StickyImageWithText.propTypes = {
  data: PropTypes.shape({
    content: PropTypes.shape({
      image: PropTypes.string.isRequired,
      paragraphs: PropTypes.arrayOf(PropTypes.string).isRequired,
      quote: PropTypes.string.isRequired,
      imageLeft: PropTypes.bool.isRequired
    })
  }).isRequired
};

export default StickyImageWithText;
