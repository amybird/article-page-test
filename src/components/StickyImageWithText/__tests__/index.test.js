import React from 'react';
import renderer from 'react-test-renderer';
import StickyImageWithText from '../index';
import { data } from '../../../data/article';

describe('Container', () => {
  it('should render a container', () => {
    const tree = renderer
      .create(<StickyImageWithText data={data.content[1]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
