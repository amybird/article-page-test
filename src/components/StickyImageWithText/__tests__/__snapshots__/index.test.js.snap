// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render a container 1`] = `
.c2 {
  font-family: serif;
  line-height: 26px;
  font-size: 20px;
  margin: 20px 0;
}

.c2::first-letter {
  font-size: 50px;
}

.c3 {
  font-family: serif;
  line-height: 26px;
  font-size: 20px;
  margin: 20px 0;
}

.c3::first-letter {
  font-size: 20px;
}

.c0 {
  width: 100%;
  max-width: 1080px;
  margin: 0 auto;
}

.c4 {
  font-size: 30px;
  line-height: 40px;
  width: 100%;
  padding: 10px;
  text-align: left;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  width: 100%;
  top: 5px;
  display: inline-block;
}

.c1 {
  width: 100%;
  display: inline-block;
  padding: 20px 10px;
}

@media (min-width:768px) {
  .c4 {
    font-size: 40px;
    line-height: 50px;
  }
}

@media (min-width:768px) {
  .c5 {
    margin-top: 70px;
    padding: 20px;
    position: -webkit-sticky;
    position: sticky;
    width: 50%;
    vertical-align: top;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 50%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      Going unnoticed is not in Danai Gurira’s repertoire. When the 41-year-old actress walks into the roof restaurant of the Dream hotel in Hollywood, wearing faded denim, a cream silk blouse and mirrored aviators, she practically gleams in the morning light. It’s hard to say if the other patrons recognize her as General Okoye, the fierce protector of the Wakanda royal family in Black Panther, or the equally imposing zombie-slayer Michonne from The Walking Dead, or are just struck by the way she carries herself, but all eyes are on her.
    </p>
    <p
      className="c3"
    >
      “I’m running a writers’ room near here,” Gurira says, sitting down and peering over the railing at the sprawl of surrounding offices and studios. She is talking about the newest line on her résumé: head writer for a television mini-series. Aside from her commanding performances on screen, she’s also a feted writer whose plays have garnered multiple awards, including a Tony nomination. Now she’s adapting Chimamanda Ngozi Adichie’s acclaimed novel Americanah – a love story following a Nigerian couple through their complicated lives – with her longtime friend Lupita Nyong’o. The two became close in 2015 when Nyong’o starred in Gurira’s all-female play Eclipsed, about the Liberian civil war.
    </p>
    <p
      className="c4"
    >
      “It’s about collaborating NOT cooperating. I’m pretty willing to have a CLEAR conversation about what I think WORKS, and what doesn’t, and why”
    </p>
  </div>
  <div
    className="c5"
  >
    <picture>
      <img
        alt="this is a lovely image"
        src="https://cache.net-a-porter.com/content/images/story-body-content-SUBBED-0-1556266075382.jpeg/w950_q65.jpeg"
      />
    </picture>
  </div>
</div>
`;
